platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/events/PromptOutputEvent.java

```diff
@@ -41,7 +42,25 @@ public void render(StyledTextOutput output) {
     /**
      * Converts the given text into the response object, or returns a new prompt to display to the user.
      */
-    public abstract Either<?, String> convert(String text);
+    public abstract PromptResult<?> convert(String text);
```

Comment:
❓ What was wrong with `Either`?

Reply:
It lives in `:functional`, which require Java 8

Reply:
`Optional` is part of its interface so it cannot be easily moved out.